overlord/hookstate/ctlcmd/set.go

```diff
@@ -42,6 +56,51 @@ func (s *setCommand) Execute(args []string) error {
 		return fmt.Errorf("cannot set without a context")
 	}
 
-	// TODO: Talk to the handler to take care of the set request.
+	// Extract the transaction from the context. If none, make one.
+	s.context().Lock()
+	transaction, ok := s.context().Cached("transaction").(*configstate.Transaction)
+	s.context().Unlock()
+	if !ok {
```

Comment:
That's making the cache lock pull double-duty, but alright. I've made the context.Lock() apply to all data in the context, and put all of these operations inside the lock.